### DISCUSSION TITLE
single function to load targets from any working dir

### DISCUSSION THREAD

**Discussion by yonicd** [ropensci/targets#885] [Help]
Is there a simple way to run something similar to this?

```r
withr::with_dir(rprojroot::find_root('_targets.R'),{
  targets::tar_load(c('object_a', 'object_b'))
})
```

I use this a lot from within (R|Q)mds so i can run the knit from any subdir and still have the ability to run it interactively or from a child session.

It would be really convenient if there is a way to do this without writing out the wrapper every time.

**Comment by wlandau** [comment]
You can use the `store` argument of `tar_load()`, e.g. `tar_load(your_target, store = "../../_targets/")`. That should be fine if the data store you supply is the data store of the pipeline that has `tar_render()` or `tar_quarto()` on that document.

**Reply by yonicd** [reply]
The only problem is that i want it totally generic, ../../ assumes you know where root is

**Reply by wlandau** [reply]
This is the sort of situation that requires custom code to find the root. {targets} assumes you stay at the root for the whole pipeline, reports that run from subdirectories are an edge case.

**Reply by andrewheiss** [reply]
This thread was really helpful! I ended up using `tar_config_set()` to set the store argument for the whole session, like this:

```r
library(targets)
tar_config_set(store = here::here('_targets'))

tar_read(some_target_name)
tar_read(some_other_target)
```

**Comment by jameelalsalam** [comment]
You may already be aware of the quarto project option execute-dir: project
https://quarto.org/docs/reference/projects/core.html

This gets at the same general issue for me. It allows me to put .qmd in a subdirectory, while having them run with working directory at the project root (thus matching the targets project root).

Personally I have not updated my version of Rstudio recently, so for some reason this is working for 'book' type quarto projects for me right now, but not regular projects. But I believe based on [this issue](https://github.com/rstudio/rstudio/issues/11361) that is fixed in the latest Rstudio release.

I suppose if you want other pieces of code within the qmd to run from the subdirectory, then this would not solve the issue for you.
